### ISSUE TITLE
alpine cannot build rust agent

### ISSUE THREAD

**Issue by yyyeerbo** [kata-containers/osbuilder#386]
# alpine doesn't support x86_64-unknown-linux-gnu
# It only support x86_64-unknown-linux-musl. Even worse,
# it doesn't support proc-macro, which is needed for serde_derive

**Comment by liubin** [comment]
https://github.com/rust-lang/rust/pull/69519

**Comment by liubin** [comment]
May be fixed by 1.44 https://github.com/rust-lang/rust/issues/40174#issuecomment-618999438

**Comment by liubin** [comment]
I have tried this and verified that 1.45 has resolved this issue.

```
$ docker run -it --rm -v `pwd`/kata-containers:/kc  rust:1.45.2-alpine3.12 sh
# cd /kc
# apk add --no-cache musl-dev
# cargo build
```
